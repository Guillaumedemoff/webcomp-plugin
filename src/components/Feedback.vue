<template>
    <div>
        <b-alert v-if="data.status == 'success'" show variant="info">{{data.output}}</b-alert>
        <b-alert v-else-if="data.status == 'unreachable'" show variant="danger">Chould not reach pythia</b-alert>
        <b-alert v-else show variant="danger">Error: {{data.status}}</b-alert>
    </div>
</template>
<script>
export default {
    name: "Feedback",
    props: {
        data: {status: String}
    }
}
</script>