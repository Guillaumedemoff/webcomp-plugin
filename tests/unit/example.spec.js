import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'

import Feedback from '@/components/Feedback.vue'

const localVue = createLocalVue()
localVue.use(BootstrapVue)

describe('Feedback.vue', () => {
  it('should render correctly', () => {
    
    const wrapper = mount(Feedback,
      {
        localVue,
        propsData: { data: {} }
      })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('does not render anything if data is empty', () => {
    const data = {}
    const wrapper = shallowMount(Feedback, {
      localVue,
      propsData: { data }
    })
    expect(wrapper.text()).toMatch("")
  })
})
