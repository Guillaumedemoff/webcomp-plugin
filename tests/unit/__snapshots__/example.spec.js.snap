// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback.vue should render correctly 1`] = `
<div>
  <div
    aria-atomic="true"
    aria-live="polite"
    class="alert alert-danger"
    role="alert"
  >
    <!---->
    Error: 
  </div>
</div>
`;
